Don't pick up libatomic from ports gcc if it happens to be installed.

Index: configure
--- configure.orig
+++ configure
@@ -17133,42 +17133,6 @@ ac_compile='$CXX -c $CXXFLAGS $CPPFLAGS conftest.$ac_e
 ac_link='$CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether to use -latomic" >&5
-$as_echo_n "checking whether to use -latomic... " >&6; }
-oLDFLAGS="$LDFLAGS"
-LDFLAGS="$LDFLAGS -Wl,--as-needed -latomic"
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <atomic>
-    int main()
-    {
-        static std::atomic<int> a{0};
-        a = a.fetch_add(1);
-        return 0;
-    }
-
-_ACEOF
-if ac_fn_cxx_try_link "$LINENO"; then :
-  qpdf_USE_ATOMIC=1
-else
-  qpdf_USE_ATOMIC=0
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-if test "$qpdf_USE_ATOMIC" = "1"; then
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-  LDFLAGS="$oLDFLAGS"
-fi
-ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
-ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
-ac_compiler_gnu=$ac_cv_c_compiler_gnu
-
 
 if test "x$qpdf_OS_SECURE_RANDOM" = "x1"; then
   OLIBS=$LIBS
